user-console/src/main/java/org/pentaho/mantle/client/solutionbrowser/SolutionBrowserPanel.java

```diff
@@ -392,6 +395,17 @@ public static String pathToId( String path ) {
     return NameUtils.URLEncode( id );
   }
 
+  private String encodeGenericFilePath( String path ){
+    return NameUtils.URLEncode( encodePath(path) );
+  }
+
+  //TODO use GenericFilenameUtils.encodePath( String path ) instead. Hitting build issue right now on @NonNull and @Nullable annotations
```

Comment:
I'd make a note in the hoth channel and add the complete error log and/or compilation error. The aim is to start conversation on the limitations on those annotations (`@NonNull @Nullable`) in GWT.

Reply:
This SDM problem has been resolved via some dependency changes [here](https://github.com/pentaho/pentaho-platform/pull/5616/files#diff-b95754c647cdb0e7a2f6678afd4e031dedaccedaf020d7b2b5e43f5bdcddde46R54) and https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1039